scipy/optimize/_lsap_module.c

```diff
@@ -56,19 +59,39 @@ calculate_assignment(PyObject* self, PyObject* args)
     int num_rows = PyArray_DIM(obj_cont, 0);
     int num_cols = PyArray_DIM(obj_cont, 1);
 
+    // test for NaN and -inf entries
+    for (size_t i=0;i<(size_t)num_rows*num_cols;i++) {
+        if (cost_matrix[i] != cost_matrix[i] || cost_matrix[i] == -INFINITY) {
+            PyErr_SetString(PyExc_ValueError,
+                            "matrix contains invalid numeric entries");
+            goto cleanup;
+        }
+    }
+
     npy_intp dim[1] = { num_rows };
-    x = PyArray_SimpleNew(1, dim, NPY_INT64);
-    int result = solve_rectangular_linear_sum_assignment(
-      num_rows, num_cols, cost_matrix, PyArray_DATA((PyArrayObject*)x));
-    if (result != 0) {
+    a = PyArray_SimpleNew(1, dim, NPY_INT64);
```

Comment:
I have added error checking for `a`, `b`, and `result`. I have made them raise `PyExc_MemoryError`, which I think is the most appropriate exception type?

Reply:
It should follow the normal python convention of setting the exception internally. So, I would just let that exception propagate:
```c
a = PyArray_SimpleNew(1, dim, NPY_INT64);
if (!a) goto cleanup;
```

Also note that you can't do:
```c
a = PyArray_SimpleNew(1, dim, NPY_INT64);
b = PyArray_SimpleNew(1, dim, NPY_INT64);

if (!a || !b) goto cleanup;
```
because the second function might clobber the error state.

Reply:
Thanks for that. I have pushed a fix.